### ISSUE TITLE
Rules: Add `mkdocstrings` directive rule for `::: identifier` blocks

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#362]
## Problem

`mkdocstrings` uses a non-CommonMark directive syntax that looks like this:

```markdown
:::
    handler: python
    options:
      show_source: false
```

The indented YAML is required for `mkdocstrings`, but it can look like an indented code block to generic Markdown tooling. `rumdl` currently has no dedicated rule that understands these blocks, so linting and autofix can end up treating the YAML as normal Markdown content instead of `mkdocstrings` configuration.

## Proposal

Add a new rule dedicated to `mkdocstrings` directives (MkDocs flavor).

Scope:
- Recognize `mkdocstrings` blocks that start with `::: <identifier>`.
- Treat the following indented YAML block as directive configuration (not a code block).
- Keep these blocks stable under `rumdl fmt` and `rumdl check --fix` (do not rewrite the YAML into fenced code blocks, do not reflow it).

## Expected behavior

Given:

```markdown
:::
    options:
      show_source: false
```

`rumdl fmt` should preserve the directive structure and indentation, and linting should not flag the YAML as a normal indented code block.

## References

- https://github.com/mkdocstrings/mkdocstrings
- https://github.com/mkdocstrings/python
- https://zensical.org/docs/setup/extensions/mkdocstrings

**Comment by rvben** [comment]
Thanks for the suggestion @pygarap!

This is fully supported. mkdocstrings blocks (`::: identifier` followed by indented YAML) are:

- Detected via `is_autodoc_marker()` and `is_autodoc_options()`
- Tracked with `in_mkdocstrings` flag in `LineInfo`
- Skippable via `filtered_lines().skip_mkdocstrings()` API
- Preserved by `rumdl fmt` and `rumdl check --fix`

Rules like MD007, MD013, and MD046 correctly skip mkdocstrings content.
